test/test_autograd.py

```diff
@@ -2466,6 +2466,19 @@ def test_rnn_backward_to_input_but_not_parameters_cuda(self):
         out.sum().backward()
         self.assertFalse(s.grad is None or s.grad.abs().sum().item() == 0)
 
+    @unittest.skipIf(not torch.cuda.is_available(), "CUDA unavailable")
+    @skipIfRocm
+    def test_lstmcell_backward_only_one_output_grad(self):
+        # checks that undefined gradients doen't hamper the backward
+        # see #11872
+        dev = torch.device('cuda')
+        l = torch.nn.LSTMCell(2, 3).to(dev).double()
+        s = torch.randn(1, 2, device=dev, dtype=torch.double, requires_grad=True)
+        for i in range(2):
+            out = l(s)[i]
+            out.sum().backward()
+            self.assertFalse(s.grad is None or s.grad.abs().sum().item() == 0)
```

Comment:
This isn't that safe. You can very well get 0 grad if RNG doesn't treat you well. I think just checking for non-throwing would be fine (maybe with a comment). 

If you really to, you can also make `s` positive, initialize lstm weights to identity, bias to zeros, and use relu, to make sure that there is positive gradient (you can check for exact values too). But that might be too much work.

Reply:
Is it really unsafe? I thought (and tested) that we have fixed random in the tests.
The assert is copied from the test above which has not seen complaints as far as I know.

Reply:
Where is the RNG? This is just a cell and not a whole network

Reply:
cell weights and `s`.